### ISSUE TITLE
Incorrect unparsing of Map cell collections

### ISSUE THREAD

**Issue by ttuegel** [runtimeverification/k#1135]
This Kore term:

```
\forall{SortGeneratedTopCell{}}(
    x:SortStateCell{},
    \ceil{SortStateCellMap{}, SortGeneratedTopCell{}}(
        Lbl'Unds'StateCellMap'Unds'{}(
            LblStateCellMapItem{}(
                Lbl'-LT-'slot'-GT-'{}(
                    Lbl'UndsStar'Int'Unds'{}(
                        Lbl'Unds'-Int'Unds'{}(
                            Lbl'UndsSlsh'Int'Unds'{}(
                                VarSlot:SortInt{},
                                \dv{SortInt{}}("4")
                            ),
                            \dv{SortInt{}}("1")
                        ),
                        \dv{SortInt{}}("4")
                    )
                ),
                x:SortStateCell{}
            ),
            VarDotVar2:SortStateCellMap{}
        )
    )
)
```

is incorrectly unparsed by `kprove` as `#Forall x . #Ceil ( DotVar2 x )`.

**Comment by ttuegel** [comment]
@dwightguth The context here is that the Haskell backend would like to abstract over Map values when simplifying Ceils, because of #1334. I guess cell collections use the entire cell as the value, which contains the key, so the frontend doesn't unparse the key separately, is that about right?

**Comment by dwightguth** [comment]
that's correct, we discard the key because the assumption is that the key is equal to the first child of the value.

**Comment by ttuegel** [comment]
Fixed in kframework/kore#1805.
